generators/server/index.js

```diff
@@ -638,10 +645,20 @@ module.exports = JhipsterServerGenerator.extend({
                 if (this.authenticationType === 'session') {
                     this.rememberMeKey = crypto.randomBytes(20).toString('hex');
                 }
-                if (this.authenticationType === 'jwt') {
+
+                if (this.authenticationType == 'jwt' || this.applicationType == 'microservice' || this.applicationType == 'uaa') {
                     this.jwtSecretKey = crypto.randomBytes(20).toString('hex');
                 }
 
+                //this now is job for uaa
+                if(this.applicationType == 'gateway') {
```

Comment:
can we do this in `app/index.js` itself similar to how we set it for microservice